index.js

```diff
@@ -0,0 +1,34 @@
+const express = require('express');
+
+// define environment variables
+const PORT = process.env.PORT || 3000;
+const HOOK = process.env.HOOK || '/hook123456789';
+
+// create express app
+const app = express();
+
+let updateRunning = false;
+
+app.get(HOOK, (req, res) => {
+  // avoid to start multiple concurrent updates
+  if (!updateRunning) {
+    updateRunning = true;
+    const { exec } = require('child_process');
+    exec('./update.sh', (err, stdout, stderr) => {
+      if (err) {
+        console.error(err);
+      } else {
+        console.log(stdout);
+      }
+      updateRunning = false;
```

Comment:
I don't think `updateRunning = false;` will be executed if `exec('./update.sh')` crashes. Might want to wrap the script execution in a try-catch block to be safe. Then again you have nodemon around it, so technically it would restart, but you might miss updates in the meantime, and I'd say it's cleaner to handle that case in the application.

Reply:
that's fair.